pkg/cli/logfile/storage.go

```diff
@@ -95,3 +108,78 @@ func read(r io.Reader) ([]Entry, error) {
 
 	return entries, nil
 }
+
+// Span is a type similar to otel's SpanStum, but with the correc types needed
+// for handle marshaling and unmarshaling.
+type Span struct {
+	Name                   string
+	SpanContext            trace.SpanContext
+	Parent                 trace.SpanContext
+	SpanKind               trace.SpanKind
+	StartTime              time.Time
+	EndTime                time.Time
+	Attributes             []attribute.KeyValue
+	Events                 []tracesdk.Event
+	Links                  []tracesdk.Link
+	Status                 tracesdk.Status
+	DroppedAttributes      int
+	DroppedEvents          int
+	DroppedLinks           int
+	ChildSpanCount         int
+	Resource               []attribute.KeyValue
+	InstrumentationLibrary instrumentation.Library
```

Comment:
nit: Comments for each export, I know some of this is unclear so feel free to reflect that ambiguity here but just nice to have something

Reply:
May also want to justify this duplication for future readers so they know the struggle

Reply:
I did add some justification at the top of the type and annotated the field that we needed to change. I feel like this represents the correct level of ambiguity since not even the OpenTelemetry library itself has any information about these fields https://pkg.go.dev/go.opentelemetry.io/otel/sdk@v1.11.0/trace/tracetest#SpanStub

Reply:
I don't feel too strongly about this, however, so I can definitely add it if you think it's necessary. I just didn't want to make descriptors for library fields that we don't own and did not define.

Reply:
Sounds reasonable!